solution_template/spinnaker/mainTemplate.json

```diff
@@ -0,0 +1,390 @@
+      },
+      "type": "string"
+    },
+    "publicIPName": {
+      "metadata": {
+        "description": "Public IP name"
+      },
+      "type": "string"
+    },
+    "publicIPNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' if the public IP has to be created"
+      },
+      "type": "string"
+    },
+    "publicIPResourceGroup": {
+      "metadata": {
+        "description": "The Public IP Resource Group"
+      },
+      "type": "string"
+    },
+    "storageAccountName": {
+      "metadata": {
+        "description": "Storage Account Name"
+      },
+      "type": "string"
+    },
+    "storageAccountNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' or 'existing'"
+      },
+      "type": "string"
+    },
+    "storageAccountType": {
+      "defaultValue": "Standard_LRS",
+      "metadata": {
+        "description": "Storage Account Type"
+      },
+      "type": "string"
+    },
+    "storageAccountResourceGroup": {
+      "metadata": {
+        "description": "The Storage Account Resource Group"
+      },
+      "type": "string"
+    },
+    "vmName": {
+      "metadata": {
+        "description": "Virtual Machine Name (also used as a prefix for other resources)"
+      },
+      "type": "string"
+    },
+    "vmSize": {
+      "metadata": {
+        "description": "Virtual Machine Size"
+      },
+      "type": "string"
+    },
+    "location": {
+      "metadata": {
+        "description": "Azure location where to deploy the resources"
+      },
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]"
+    }
+  },
+  "variables": {
+    "_extensionScript": "101-spinnaker.sh",
+    "frontEndNSGName": "[concat(parameters('vmName'), 'NSG')]",
+    "nestedUrl": "[concat(parameters('_artifactsLocation'),'/solution_template/nested')]",
+    "nicName": "[concat(parameters('vmName'), 'VMNic')]",
+    "OSDiskName": "[concat(parameters('vmName'), 'OSDisk-', uniqueString(resourceGroup().id, deployment().name))]",
+    "publicIPDeploymentName": "[concat(parameters('publicIPName'), '-publicIP-deploy')]",
+    "storageDeploymentName": "[concat(parameters('storageAccountName'), '-storage-deploy')]",
+    "subnetName": "[concat(parameters('vmName'), 'Subnet')]",
+    "subnetPrefix": "10.0.0.0/24",
+    "templateUrlIP": "[concat(variables('nestedUrl'),'/publicIP-',parameters('publicIPNewOrExisting'),'.json',parameters('_artifactsLocationSasToken'))]",
+    "templateUrlStorage": "[concat(variables('nestedUrl'),'/storageAccount-',parameters('storageAccountNewOrExisting'),'.json',parameters('_artifactsLocationSasToken'))]",
+    "templateUrlVM": "[concat(variables('nestedUrl'),'/VM-',parameters('authenticationType'),'.json',parameters('_artifactsLocationSasToken'))]",
+    "templateUrlVMInit": "[concat(variables('nestedUrl'), '/spinnaker-VM-Init.json', parameters('_artifactsLocationSasToken'))]",
+    "ubuntuSku": "14.04.5-LTS",
```

Comment:
According to https://www.spinnaker.io/setup/install/halyard/

"Currently Halyard may only be installed on Ubuntu 14.04"